src/mlx/warnings_checker.py

```diff
@@ -124,22 +124,27 @@ def return_check_limits(self):
 
         Returns:
             int: 0 if the amount of warnings is within limits, the count of warnings otherwise
+                (or 1 in case of a count of 0 warnings)
         '''
+        error_code = self.count
         if self.warn_min == self.warn_max and self.count == self.warn_max:
             print("Number of warnings ({0.count}) is exactly as expected. Well done."
                   .format(self))
+            return 0
         elif self.count > self.warn_max:
-            print("Number of warnings ({0.count}) is higher than the maximum limit ({0.warn_max}). "
-                  "Returning error code 1.".format(self))
-            return self.count
+            error_reason = "higher than the maximum limit"
         elif self.count < self.warn_min:
-            print("Number of warnings ({0.count}) is lower than the minimum limit ({0.warn_min}). "
-                  "Returning error code 1.".format(self))
-            return self.count
+            error_reason = "lower than the minimum limit"
         else:
             print("Number of warnings ({0.count}) is between limits {0.warn_min} and {0.warn_max}. Well done."
                   .format(self))
-        return 0
+            return 0
+        # code below is only reached after failure; don't allow error code 0
```

Comment:
This is an ugly solution, but at this moment can't think of more elegant.

Reply:
Goto does not exist but functions do. Maybe we put below printout to function and then call return with function call which would return proper error value?

Reply:
I agree that it was ugly. I added a function as you suggested: 1810e3bb21f0cbcc6a0427f409c7d3c87c8fee03